frontend/taipy-gui/src/components/Taipy/ProgressBar.tsx

```diff
@@ -0,0 +1,137 @@
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import {
+    Box,
+    CircularProgress,
+    CircularProgressProps,
+    LinearProgress,
+    LinearProgressProps,
+    Typography,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+
+interface ProgressBarProps {
+    linear?: boolean; //by default - false
+    showProgress?: boolean; //by default - false
+}
+
+const Progress = (props: ProgressBarProps) => {
+    const { linear, showProgress } = props;
+
+    const [linearProgress, setLinearProgress] = useState(false);
+    const [progressVisible, setProgressVisible] = useState(false);
+
+    useEffect(() => {
+        setLinearProgress((progress) => {
+            return props.linear !== undefined ? props.linear : progress;
+        });
+    }, [props.linear, linear]);
+
+    useEffect(() => {
+        setProgressVisible((progress_visible) => {
+            return props.showProgress !== undefined ? props.showProgress : progress_visible;
+        });
+    }, [props.showProgress, showProgress]);
+
+    //circular progress element
+    function CircularProgressWithLabel(props: CircularProgressProps & { value: number }) {
+        return (
+            <Box sx={{ position: "relative", display: "inline-flex" }}>
+                <CircularProgress variant="determinate" {...props} />
+                <Box
+                    sx={{
+                        top: 0,
+                        left: 0,
+                        bottom: 0,
+                        right: 0,
+                        position: "absolute",
+                        display: "flex",
+                        alignItems: "center",
+                        justifyContent: "center",
+                    }}
+                >
+                    <Typography variant="caption" component="div" color="text.secondary">
+                        {`${Math.round(props.value)}%`}
+                    </Typography>
+                </Box>
+            </Box>
+        );
+    }
+
+    //render circular progress element
+    function CircularWithValueLabel() {
+        const [progress, setProgress] = useState(10);
+
+        useEffect(() => {
+            const timer = setInterval(() => {
+                setProgress((prevProgress) => (prevProgress >= 100 ? 0 : prevProgress + 10));
+            }, 800);
+            return () => {
+                clearInterval(timer);
+            };
+        }, []);
+        return <CircularProgressWithLabel value={progress} />;
+    }
+
+    //linear progress element
+    function LinearProgressWithLabel(props: LinearProgressProps & { value: number }) {
```

Comment:
I think the   `LinearProgressWithLabel`  & `CircularProgressWithLabel` are enough.

Reply:
Yes, those functions are enough, but to render the elements with label, we need a function which passes some params(like - the progress interval and the timeout function) to these 2 functions. These were given in the Mui official doc. Otherwise, we would have to add those ourselves.

Reply:
we don't want progress interval of timeout functions

Reply:
Got it👍🏼 I've removed both the functions containing the timeout or progress intervals and done it like this. Is this okay?<br>
![image](https://github.com/Avaiga/taipy/assets/129659514/d382fb63-6e77-4bd3-8565-5f77699d8cbe)

Reply:
you should use the value declared in factory.py
you'll get it automatically has 2 properties of the component:
defaultValue?: number
value?: number

you can use a taipy-gui hook 
const value = useDynamicProperty(props.value, props.defaultValue, 0);

Reply:
Ahh! looks like I pushed the changes too early. I'll make the changes for this👍🏼

Reply:
In [the above comment](https://github.com/Avaiga/taipy/pull/1303#discussion_r1611438837), if I'm right, then we need 2 params, right? Like we can declare the `defaultValue` in the `factory.py` (at the top where all the default params are declared) and then keep the `value` param as optional for the user to enter the % ?

Reply:
no when you declare a prop_x property as dynamic* in the factory.py, you get 2 properties defaultPropX and propX on your front end component

Reply:
Actually, I was trying to understand the `useDynamicProperty` hook. As far as I've understood, I've to create 2 new vars in the `ProgressBarProps` i.e. `value` and `defaultValue` and then the `useDynamicProperty` hook has to be used so that if the user enters the value, then that will be used, otherwise 0. Am I correct? Moreover, in the `factory.py`, the `value` var has to be added(which I've already done) - https://github.com/Avaiga/taipy/pull/1303/files#diff-fad865192c2666416116521a44fa3ca4f17e15fbb1509d4227a5cbb2db8f1fceR53 ?

Reply:
Hey! actually, when I was trying to implement the `useDynamicProperty` hook, I'm the type error while rendering. I've declared the vars as (in the ProgressBarProps) - 
```Typescript
value?: number;
defaultValue?: number;
```
and then I declared the `const val = useDynamicProperty(props.value, props.defaultValue, 0);` <br>
![image](https://github.com/Avaiga/taipy/assets/129659514/7efc1dc1-a6dc-4920-9252-477e6de61337)

Reply:
PS: I don't think screenshots of code are better than vscode/github code links :-)

Reply:
I've just pushed some code with some changes. But, the error is still there(the type error, see ss above), could you please guide how to fix this? Also, in the `factory.py`, I've made some changes in `set_value_and_default`, could you please review those as well, if something is wrong?

Reply:
> You can force the type of value to `number value as number`

Reply:
Done, I've changed `value={val}` to `value={val as number}`. Is this correct?